torch/csrc/jit/codegen/cuda/lower_double_buffer.cpp

```diff
@@ -568,6 +912,74 @@ void DoubleBufferInfo::build(Fusion* fusion) {
         GpuLower::current()->caMap()->getConcreteMappedID(
             double_buffer_axis, IdMappingMode::LOOP));
   }
+
+  // Add a second pass to keep track of lifted
+  //  double buffer loop nest see also [Skew Double Buffer Loop Transformation].
+  for (auto& info : map_) {
+    buildSkewInfo(info.first, info.second);
+  }
+}
+
+void DoubleBufferInfo::buildSkewInfo(
+    const TensorView* tv,
+    const TvInfo& tv_info) {
+  if (tv->shouldSkewDoubleBuffer()) {
+    // Detect the outer main loop
+    IterDomain* outer_loop_id = nullptr;
+    bool double_buffer_axis_found = false;
+    for (auto id_it = tv->domain()->domain().rbegin();
+         id_it != tv->domain()->domain().rend();
+         id_it++) {
+      // The outer loop to lift prolog out of would
+      //  be the first serial loop on the left of
+      //  the double buffer loop
+      if (double_buffer_axis_found &&
+          (*id_it)->getParallelType() == ParallelType::Serial) {
+        outer_loop_id = *id_it;
```

Comment:
Don't we need to check if that domain is the double buffer loop of the producer?